src/frontend/src/observer/observer_manager.rs

```diff
@@ -40,18 +42,23 @@ pub(crate) struct ObserverManager {
     worker_node_manager: WorkerNodeManagerRef,
     catalog: Arc<RwLock<Catalog>>,
     catalog_updated_tx: Sender<CatalogVersion>,
+    user_info_manager: Arc<RwLock<UserInfoManager>>,
+    user_info_updated_tx: Sender<UserInfoVersion>,
     hummock_snapshot_manager: HummockSnapshotManagerRef,
 }
 
 const RE_SUBSCRIBE_RETRY_INTERVAL: Duration = Duration::from_millis(100);
 
 impl ObserverManager {
+    #[allow(clippy::too_many_arguments)]
```

Comment:
May consider using a struct to pass these arguments. Just for readability. :)

Reply:
Can't find a suitable struct naming so far. 🤔 NTFS.